packages/next-swc/crates/next-dev-tests/tests/integration.rs

```diff
@@ -184,6 +226,21 @@ async fn run_test(resource: PathBuf) -> JestRunResult {
     );
 
     let package_root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let tests_dir = package_root.join("tests");
+    let integration_tests_dir = tests_dir.join("integration");
+    let resource_temp: PathBuf = tests_dir.join("temp").join(
```

Comment:
The HMR test modify the source code from the test. So to avoid modifying the origin (commited) source code, it's copied to a temp directory and can be modified there. This avoids having weird changes in the source code when tests are cancelled or exit in a unexpected way.